arroyo/processing/processor.py

```diff
@@ -156,10 +156,14 @@ def __commit(
         be used during consumer shutdown where we do not want to wait before committing.
         """
         self.__consumer.stage_positions(positions)
-        self.__messages_since_last_commit += 1
+
+        messages_since_last_commit = 0
+        for partition, pos in positions.items():
+            messages_since_last_commit += pos.offset
+            messages_since_last_commit -= self.__comitted_offsets.get(partition, 0)
```

Comment:
Yeah. I think I fixed that in a more recent commit. I couldn't adopt your logic because, since at the start of the consumer there is nothing in `__committed_offsets`, the `messages_since_last_commit` will be `0`, so the commit policy will return false, ensuring that this map continues to stay empty. This leads to the consumer never committing.

I now assume that the first offset-to-be-committed equals to an offset delta of 1 (this seems just as wrong though?), and if there's nothing in the map I write the previous offset back even if I don't commit.

Reply:
Thanks. I think assuming it's 1 is reasonable since attempting to fetch anything more accurate is something that we definitely don't want to do here.